test/Common.cpp

```diff
@@ -163,9 +163,92 @@ bool CommonOptions::parse(int argc, char const* const* argv)
 	po::variables_map arguments;
 	addOptions();
 
+	std::vector<const char*> args;
+	string importPath = "semanticTests/importedTests";
+
+	for(uint i = 0; i < (uint)argc; i++)
+	{
+		if(((string)argv[i] == "-t" || (string)argv[i] == "--test") && (uint) argc - 1 > i)
+		{
+			string testPath = solidity::test::testPath().lexically_normal().string();
+			if (boost::filesystem::exists(argv[i+1]) && !strstr(argv[i+1], testPath.c_str()))
+			{
+				boost::filesystem::path p(argv[i+1]);
+				boost::filesystem::path newPath = solidity::test::testPath().lexically_normal();
+				newPath /= "/libsolidity/";
+				newPath /= importPath;
+				if (!boost::filesystem::exists(newPath.lexically_normal()))
+				{
+					boost::filesystem::remove_all(newPath);
+				}
+				if (boost::filesystem::is_regular_file(argv[i+1]))
+				{
+
+					if(boost::filesystem::create_directory(newPath.lexically_normal()))
+					{
+						newPath /= p.filename();
+
+						if(boost::filesystem::extension(newPath) == ".sol" || boost::filesystem::extension(newPath) == ".yul")
+						{
+							boost::filesystem::copy_file(argv[i+1], newPath.c_str());
+						}
+
+						args.push_back(argv[i]);
+						importPath += "/";
+						importPath += p.filename().string();
+						auto importIndex = importPath.find_last_of(".");
+						importPath = importPath.substr(0, importIndex);
+						args.push_back(importPath.c_str());
+						i++;
+					}
```

Comment:
Copying the test file just to run it feels like weird, round-about way to implement this. And will make it impossible to run the test suite if the directory is not writable. That's a big and unnecessary assumption.

A better way to do this would be to register the tests to run under some special name (you could give them some prefix to distinguish them from tests added from the usual location) and then run specifically those tests.

https://github.com/ethereum/solidity/blob/be9c48e131e5d8616c8bdc7200119b6672d5ff3f/test/soltest.cpp#L132-L202